trilinoscouplings: trilinos configuration fails when trlinoscoupings=ON but tpetra=OFF

## Bug Report

@hkthorn and Joshua reported me (@eeprude) the trilinos configuration error described below.

### Description

BEGIN COPY OF A PORTION OF THE CONFIGURATION OUTPUT INTO THE SCREEN

...
Processing enabled top-level package: TrilinosCouplings (Libs)
CMake Error at cmake/tribits/core/utils/AssertDefined.cmake:49 (message):
  Error, the variable TpetraCore_ETI_SCALARS_NO_ORDS is not defined!
Call Stack (most recent call first):
  packages/trilinoscouplings/CMakeLists.txt:24 (ASSERT_DEFINED)
 
 
CMake Error at cmake/tribits/core/utils/AssertDefined.cmake:49 (message):
  Error, the variable TpetraCore_ETI_LORDS is not defined!
Call Stack (most recent call first):
  packages/trilinoscouplings/CMakeLists.txt:25 (ASSERT_DEFINED)
 
 
CMake Error at cmake/tribits/core/utils/AssertDefined.cmake:49 (message):
  Error, the variable TpetraCore_ETI_GORDS is not defined!
Call Stack (most recent call first):
  packages/trilinoscouplings/CMakeLists.txt:26 (ASSERT_DEFINED)
 
 
CMake Error at cmake/tribits/core/utils/AssertDefined.cmake:49 (message):
  Error, the variable TpetraCore_ETI_NODES is not defined!
Call Stack (most recent call first):
  packages/trilinoscouplings/CMakeLists.txt:27 (ASSERT_DEFINED)
 
 
 
Processing explicit instantiation support for enabled packages ...
 
Processing ETI support: TrilinosCouplings
-- TrilinosCouplings: Processing ETI / test support
-- Enabled Scalar types:        
-- Enabled LocalOrdinal types:  
-- Enabled GlobalOrdinal types: 
-- Enabled Node types:          
-- Set of enabled types, before exclusions: 
...

END COPY OF A PORTION OF THE CONFIGURATION OUTPUT INTO THE SCREEN

### Steps to Reproduce

Try to configure trilinos with the following options:

...
-D Trilinos_ENABLE_Tpetra:BOOL=OFF \
-D Trilinos_ENABLE_TrilinosCouplings:BOOL=ON \
...

--- a/packages/trilinoscouplings/CMakeLists.txt
+++ b/packages/trilinoscouplings/CMakeLists.txt
@@ -10,33 +10,35 @@ TRIBITS_PACKAGE(TrilinosCouplings  DISABLE_CIRCULAR_REF_DETECTION_FAILURE)
 # B) Set up package-specific options
 #
 
-#
-# Do explicit template instantiation (ETI) and testing over the same
-# Scalar, LocalOrdinal, and GlobalOrdinal types as for Tpetra objects.
-# However, by default, exclude all of Tpetra's GlobalOrdinal (GO)
-# types from the ETI list.  This is because Tpetra adds all GO ETI
-# types to the list of Scalar ETI types.  (This lets it implement
-# certain communication patterns involving e.g., Vector<GO, LO, GO>.)
-#
+IF(HAVE_TRILINOSCOUPLINGS_TPETRA)
+  #
+  # Do explicit template instantiation (ETI) and testing over the same
+  # Scalar, LocalOrdinal, and GlobalOrdinal types as for Tpetra objects.
+  # However, by default, exclude all of Tpetra's GlobalOrdinal (GO)
+  # types from the ETI list.  This is because Tpetra adds all GO ETI
+  # types to the list of Scalar ETI types.  (This lets it implement
+  # certain communication patterns involving e.g., Vector<GO, LO, GO>.)
+  #
 
-# Make sure that Tpetra actually defined these variables, even if they
-# are empty.
-ASSERT_DEFINED(TpetraCore_ETI_SCALARS_NO_ORDS)
-ASSERT_DEFINED(TpetraCore_ETI_LORDS)
-ASSERT_DEFINED(TpetraCore_ETI_GORDS)
-ASSERT_DEFINED(TpetraCore_ETI_NODES)
+  # Make sure that Tpetra actually defined these variables, even if they
+  # are empty.
+  ASSERT_DEFINED(TpetraCore_ETI_SCALARS_NO_ORDS)
+  ASSERT_DEFINED(TpetraCore_ETI_LORDS)
+  ASSERT_DEFINED(TpetraCore_ETI_GORDS)
+  ASSERT_DEFINED(TpetraCore_ETI_NODES)
 
-# Promote these variables to be visible outside of the "scope" of this
-# directory -- e.g., to other packages.
-GLOBAL_SET(TrilinosCouplings_ETI_SCALARS ${TpetraCore_ETI_SCALARS_NO_ORDS})
-GLOBAL_SET(TrilinosCouplings_ETI_LORDS   ${TpetraCore_ETI_LORDS})
-GLOBAL_SET(TrilinosCouplings_ETI_GORDS   ${TpetraCore_ETI_GORDS})
-GLOBAL_SET(TrilinosCouplings_ETI_NODES   ${TpetraCore_ETI_NODES})
+  # Promote these variables to be visible outside of the "scope" of this
+  # directory -- e.g., to other packages.
+  GLOBAL_SET(TrilinosCouplings_ETI_SCALARS ${TpetraCore_ETI_SCALARS_NO_ORDS})
+  GLOBAL_SET(TrilinosCouplings_ETI_LORDS   ${TpetraCore_ETI_LORDS})
+  GLOBAL_SET(TrilinosCouplings_ETI_GORDS   ${TpetraCore_ETI_GORDS})
+  GLOBAL_SET(TrilinosCouplings_ETI_NODES   ${TpetraCore_ETI_NODES})
 
-TRIBITS_ADD_EXPLICIT_INSTANTIATION_OPTION()
+  TRIBITS_ADD_EXPLICIT_INSTANTIATION_OPTION()
 
-# we use this for testing as well as library eti, so do it regardless
-TRIBITS_ADD_ETI_SUPPORT()
+  # we use this for testing as well as library eti, so do it regardless
+  TRIBITS_ADD_ETI_SUPPORT()
+ENDIF()
 
 
 TRIBITS_ADD_OPTION_AND_DEFINE(${PACKAGE_NAME}_ENABLE_Newp_swahili

